clang/include/clang/Basic/Attr.td

```diff
@@ -1432,6 +1432,29 @@ def SYCLIntelMaxGlobalWorkDim : InheritableAttr {
   let SupportsNonconformingLambdaSyntax = 1;
 }
 
+def SYCLDetailDeviceGlobal: InheritableAttr {
+  let Spellings = [GNU<"device_global">,
+                   CXX11<"__sycl_detail__", "device_global">];
+  let Subjects = SubjectList<[CXXRecord], ErrorDiag>;
+  let LangOpts = [SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
+  let Documentation = [Undocumented];
+}
+
+def SYCLDetailGlobalVariableAllowed : InheritableAttr {
+  let Spellings = [GNU<"global_variable_allowed">,
+                   CXX11<"__sycl_detail__", "global_variable_allowed">];
+  let Subjects = SubjectList<[CXXRecord, GlobalStorageNonLocalVar], ErrorDiag>;
+  let LangOpts = [SYCLIsDevice, SilentlyIgnoreSYCLIsHost];
+  let Documentation = [Undocumented];
```

Comment:
```suggestion
  let Documentation = [SYCLDetailGlobalVariableAllowedAttrDocs];
```

Reply:
For attributes that the user is not supposed to use, do we not leave the the documentation field as "Undocumented"?

Reply:
Yeah, but I've seen that you added some documentation.
I was under impression that for internal attributes we don't add the documentation, but for example https://github.com/intel/llvm/pull/5535/files adds it. I think it is better to clarify it with @AaronBallman .

Reply:
I usually ask for documentation even for internal attributes; we're our own customers and having documentation on how to properly use the attribute is helpful. It also gives us a place to officially document that the attribute is an implementation detail and users should not use it themselves even if they see it in a header file, because we don't support their use of it.